File: day01/python/part2.py
with open('../input.txt') as f:
    floor = 0;
    for n, c in enumerate(f.read()):
        if floor == -1:
            print(n)
            break
        if c == '(':
            floor = floor + 1
        elif c == ')':
            floor = floor - 1
        else:
            raise Exception('Invalid input')

File: day01/haskell/part2.hs
f :: String -> Integer -> Integer
f (')':xs) 0 = 1
f ('(':xs) n = f xs (n + 1) + 1
f (')':xs) n = f xs (n - 1) + 1
f _ n = error "Invalid input"

main = readFile "../input.txt" >>= \s -> print $ f s 0
